Add Infernal Robotics Model Rework part packs

mactijn opened the pull request:
Manually generated IR Model Rework part packs.

Differs from regular distribution to be more dependency-friendly:
- Split off TweakScale support into own package, making it optional
- Split off Struts from Utility pack into their own package due to multiple dependencies
- Split the Core pack into a Robotics and a Structual package. Structural parts can now be installed without the need for IR

Unfortunately, the Struts pack is untested as the ActiveStruts download URL is currently broken

Dazpoet commented:
I fail to see which problems these changes adress. The author of the model rework have split them up into 3 different packs on KerbalStuff and as such https://github.com/KSP-CKAN/NetKAN/pull/952 and https://github.com/KSP-CKAN/NetKAN/pull/951 aim to fulfill the wish of the author about how the mod should be distributed.

If this is however a thing that should be done it belongs in NetKAN where new .ckan files would be generated every time an update happens on KS rather than in CKAN-meta where you'll have to create new .ckan files every update.

You can't have IRPartsReworkCoreRobotics provide IR-Model-Rework-Core since it does in fact not provide that mod seeing as parts of it are cut out. If another mod depends on IR-Model-Rework-Core expecting it to have structural parts and the user choose IRPartsReworkCoreRobotics the experience will not be the same.

There need to be conflicts in *all* of these files seeing as they do conflic with indexed mods. Or well not the utility pack seeing as that is still not merged. I see no dependency on CIT-utils even though that would appear to be a dependency for the utility pack.

Why make a new Expansion that is identical to the old one?

mactijn commented:
Pardon me, I completely overlooked that commit. It doesn't help that these packages didn't make it to CKAN-meta/master yet ;)

These come from a private CKAN repository I've built over the last few months, with the very specific purpose of splitting up bigger packs into small ones because of memory reasons (I'm on a mac, in-game memory is scarce).

The structural parts can be easily made part of the IR-Model-Rework-Core virtual installing experience by changing the "recommends" for them in the CoreRobotics package to a "depends". My intention was to add the option for the end user to install the structural parts without needing to have IR or any of the moving parts as an added bonus.

CIT-util is a dependency for ActiveStruts, and will be installed when IRPartsReworkRoboStruts will be installed because AS _is_ a dependency. I split them off the Utilities pack so that the other parts will not be bothered by that, if the user chooses so. Also, the downloads for AS and CIT-util are currently broken (or at least were when I tried earlier), which triggered me to publish this.

And then lastly, I split off TweakScale support, as all 3 packages install the same file to accomplish this, and thus this would cause a conflict. I made it optional, seeing as the IRR parts (struts not tested) work fine without (albeit in a "lesser" experience). 

I guess you can forego on this pull request, and I'll talk to ZodiusInfuser about splitting these packs.

Dazpoet commented:
Ah I did not know about the connection between ActiveStruts and CIT-Utils.

But yes I'll close this but if there is indeed a strong demand for something like this I recommend a PR to NetKAN which split the currently indexed files rather than creating semi-duplicates. As for why the added ones (core and expansion) aren't currently in CKAN-meta/master, that is a mystery to me aswell and I'll have someone whack the indexing bot to get them there asap.
